storage/innobase/xtrabackup/src/xbcloud/s3_ec2.cc

```diff
@@ -0,0 +1,132 @@
+/******************************************************
+Copyright (c) 2022 Percona LLC and/or its affiliates.
+
+AWS EC2 Instance Metadata client implementation using IMDSv2.
+
+More details:
+https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+
+*******************************************************/
+
+#include "xbcloud/s3_ec2.h"
+#include "msg.h"
+#include "my_rapidjson_size_t.h"
+#include "rapidjson/document.h"
+
+namespace xbcloud {
+bool S3_ec2_instance::parse_var_response(const Http_response &http_response,
+                                         std::string *var) {
+  if (http_response.body().size() == 0) {
+    return false;
+  }
+
+  if (http_response.headers().count("content-type") == 0 ||
+      http_response.headers().at("content-type") != "text/plain") {
+    return false;
+  }
+  *var = std::string(http_response.body().begin(), http_response.body().end());
+
+  return true;
+}
+
+bool S3_ec2_instance::parse_keys_response(const Http_response &http_response) {
+  if (http_response.body().size() == 0) {
+    return false;
+  }
+
+  if (http_response.headers().count("content-type") == 0 ||
+      http_response.headers().at("content-type") != "text/plain") {
+    msg_ts("%s: Failed to parse content-type for instance metadata\n",
+           my_progname);
+    return false;
+  }
+  std::string s(http_response.body().begin(), http_response.body().end());
+  rapidjson::Document document;
+  document.Parse(s.c_str());
+  if (!document.IsObject()) {
+    msg_ts("%s: Instance metadata response is not a JSON object\n",
+           my_progname);
+    return false;
+  }
+
+  if (!document.HasMember("Code") || !document.HasMember("AccessKeyId") ||
+      !document.HasMember("SecretAccessKey") || !document.HasMember("Token")) {
+    msg_ts("%s: Instance metadata response missing element\n", my_progname);
```

Comment:
can you please change print to the missing element name?

Reply:
Done